pkg/cmd/pulumi/pulumi.go

```diff
@@ -393,14 +393,28 @@ func checkForUpdate(ctx context.Context) *diag.Diag {
 		return nil
 	}
 
-	latestVer, oldestAllowedVer, err := getCLIVersionInfo(ctx)
-	if err != nil {
-		logging.V(3).Infof("error fetching latest version information "+
-			"(set `%s=true` to skip update checks): %s", env.SkipUpdateCheck.Var().Name(), err)
+	var skipUpdateCheck bool
+	latestVer, oldestAllowedVer, err := getCachedVersionInfo()
+	if err == nil {
+		// If we have cached version information then don't bother warning, we would have warned when we last
+		// updated the cache. But we will still log this to the internal logging system that by default users
+		// don't see.
+		skipUpdateCheck = true
+	} else {
+		latestVer, oldestAllowedVer, err = getCLIVersionInfo(ctx)
+		if err != nil {
+			logging.V(3).Infof("error fetching latest version information "+
+				"(set `%s=true` to skip update checks): %s", env.SkipUpdateCheck.Var().Name(), err)
+		}
 	}
 
 	if oldestAllowedVer.GT(curVer) {
-		return diag.RawMessage("", getUpgradeMessage(latestVer, curVer))
+		msg := getUpgradeMessage(latestVer, curVer)
+		if skipUpdateCheck {
+			logging.Warningf(msg)
+			return nil
+		}
```

Comment:
Am I misreading this or is the if condition here inverted?

Reply:
This is logging it to the logging. The other returns a message to print.

Reply:
Won't logging still be printed to stderr?

Reply:
Not by default.